### PR TITLE
Add test for new IncrementalMerkleTree methods

### PR DESCRIPTION
**Pull request by ebfull**

### CODE REVIEW

File: src/gtest/test_merkletree.cpp [hunk 1]
```diff
@@ -322,3 +323,22 @@ TEST(orchardMerkleTree, appendBundle) {
...
+{
+    SaplingMerkleTree tree;
+    for (size_t i = 0; i < (1 << 16); i++) {
+        EXPECT_FALSE(tree.complete_subtree_root().has_value());
+        EXPECT_EQ(tree.current_subtree_index(), 0);
+        AppendRandomLeaf(tree);
+    }
+    auto subtree_root = tree.complete_subtree_root();
+    ASSERT_TRUE(subtree_root.has_value());
+    EXPECT_EQ(tree.current_subtree_index(), 1);
```
Comment (nuttycom) [inline src/gtest/test_merkletree.cpp:337]: "current" means "will next be appended to"? Maybe this should actually be named `appendable_subtree_index` or something of the sort.

**Review by nuttycom — DISMISSED** [review]
utACK

**Review by nuttycom — DISMISSED** [review]
re-utACK https://github.com/zcash/zcash/pull/6710/commits/f8f9773806ae25aee3604bffacfad6c086e789fc

**Review by nuttycom — APPROVED** [review]
re-utACK 8bbd010d882a08a0c4a8b289aa3a83e49cce14b0

File: src/gtest/test_merkletree.cpp [hunk 2]
```diff
@@ -322,3 +323,36 @@ TEST(orchardMerkleTree, appendBundle) {
...
+    for (size_t i = 0; i < (1 << 16); i++) {
+        EXPECT_FALSE(tree.complete_subtree_root().has_value());
+        EXPECT_EQ(tree.current_subtree_index(), 0);
+        AppendRandomLeaf(tree);
+    }
+    EXPECT_EQ(tree.current_subtree_index(), 1);
+    auto subtree_root = tree.complete_subtree_root();
+    ASSERT_TRUE(subtree_root.has_value());
+    uint256 cur = *subtree_root;
+    for (size_t depth = 16; depth < 32; depth++) {
```
Comment (str4d) [inline src/gtest/test_merkletree.cpp:339]: Non-blocking nit: the `16`s in this test should be derived from `libzcash::TRACKED_SUBTREE_HEIGHT` (either directly, or as a subtraction from `SAPLING_INCREMENTAL_MERKLE_TREE_DEPTH`, as appropriate). Likewise, the `32` should be `SAPLING_INCREMENTAL_MERKLE_TREE_DEPTH`.

**Review by str4d — APPROVED** [review]
utACK with non-blocking nit.

**Review by nuttycom — APPROVED** [review]
re-utACK 6c8cc29aa53b776424e5c6290c1082398cdbe946

**Pull request merged**